### ISSUE TITLE
installation error 

### ISSUE THREAD

**Issue by optvps** [irontec/ivozprovider#1077]
Thanks for such a nice platform

I am installing artemis 2.11.1 on my fresh dedicated server 

facing installation issue with the following error

-----------------------------------------------------------------
In AbstractMySQLDriver.php line 103:
![Screenshot_24](https://user-images.githubusercontent.com/53054260/61473810-d7194900-a9a4-11e9-88b7-85d172ead9d7.jpg)

  An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused

In PDOConnection.php line 47:

  SQLSTATE[HY000] [2002] Connection refused

In PDOConnection.php line 43:

  SQLSTATE[HY000] [2002] Connection refused

doctrine:database:create [--shard SHARD] [--connection [CONNECTION]] [--if-not-exists] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command>

In AbstractMySQLDriver.php line 103:

  An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused

In PDOConnection.php line 47:

  SQLSTATE[HY000] [2002] Connection refused

In PDOConnection.php line 43:

  SQLSTATE[HY000] [2002] Connection refused

doctrine:migrations:status [--show-versions] [--configuration [CONFIGURATION]] [--db-configuration [DB-CONFIGURATION]] [--db DB] [--em EM] [--shard SHARD] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command>

In AbstractMySQLDriver.php line 103:

  An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused

In PDOConnection.php line 47:

  SQLSTATE[HY000] [2002] Connection refused

In PDOConnection.php line 43:

  SQLSTATE[HY000] [2002] Connection refused

---------------------------------------------------------------

Hope anybody help

**Comment by Kaian** [comment]
Hi @optvps

MySQL password is requested two times during the installation:

 - When mysql-server package in installed
 - In IvozProvider configuration menu

You must be sure you have entered the same password in both places during installation.

There is an install video guide in youtube, check seconds 4:42 for the first time and 5:07 for the second time.

https://www.youtube.com/watch?v=Dy3M7X6Jw5w

Regards

**Comment by optvps** [comment]
Thanks for your help !

i just created database with name "ivozprovider" and pulled data from 

/opt/irontec/ivozprovider/schema/initial.sql

and interface working now , 

it is good to see that you are using cgrates 

proxytrunk started  but showing some kind of cgrates errors

~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Jul 18 10:29:04 voice.woiko.com systemd[1]: Started Run Ivozprovider CDR mixer.
Jul 18 10:29:04 voice.woiko.com cgr-engine[3222]: CGRateS <aea5106> [3222]: Could not configure dataDb: DENIED Redis is running in protected mode because protected mode is enabled, no bind address was specified, no authentication password is requested to clients. In this mode connections are only accepted from the loopback interface. If you want to connect from external computers to Redis you may adopt one of the following solutions: 1) Just disable protected mode sending the command 'CONFIG SET protected-mode no' from the loopback interface by connecting to Redis from the same host the server is running, however MAKE SURE Redis is not publicly accessible from internet if you do so. Use CONFIG REWRITE to make this change permanent. 2) Alternatively you can just disable the protected mode by editing the Redis configuration file, and setting the protected mode option to 'no', and then restarting the server. 3) If you started the server manually just for testing, restart it with the '--protected-mode no' option. 4) Setup a bind address or an authentication password. NOTE: You only need to do one of the above things in order for the server to start accepting connections from the outside. exiting!
Jul 18 10:29:04 voice.woiko.com onError[3219]: request.INFO: Matched route "_". {"route":"_","route_parameters":{"_controller":"kernel:sync","_route":"_"},"request_uri":"http://:/","method":"GET"} []

!~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Should i contact to cgrates community for this help ?

thank you again for your help and support 
happy sips

**Comment by Kaian** [comment]
I highly recommend you to do a fresh install with everything ok, otherwise, database migrations will be broken. Appart from initial.sql dump, the package postinstall does more database changes and imports from delta migrations, that's why the mentioned configuration must be properly set.

CGRateS should connect to local redis if you've followed all steps mentioned in the video (it is for oasis, but artemis install is the same). Be sure to reboot after installation.

I don't thing CGRateS community can help here. This is an enviromental proble. I have installed yesterday an standalone iso and worked without problems. You can disable protected mode in redis but It wouldn't be necessary because you're connecting from localhost (because data.ivozprovider.local resolves to 127.0.0.1, right?)

Also, take into account that we don't offer support for installations, this forum is reserved for bugreports, feature requests and scheduling of the project. We're creating a google group so users can request information and share their experience and leave this place for project techincal issues.
It's not yet, but you can be the first one there: https://groups.google.com/forum/#!groupsettings/ivozprovider-users

Regards
